zhaquirks/tuya/tuya_trv.py

```diff
@@ -68,6 +68,25 @@ def __init__(self, *args, **kwargs):
         self.add_unsupported_attribute(Thermostat.AttributeDefs.pi_heating_demand.id)
 
 
+class TuyaThermostatV2NoSchedule(TuyaThermostatV2):
+    """Ensures schedule is disabled on system_mode change."""
+
+    async def write_attributes(self, attributes, manufacturer=None):
```

Comment:
Also, we should probably add typing here if we're adding this now. I now the underlying Tuya `write_attributes` don't have it yet, but might be nice to do it here already (and forward unspecified `**kwargs`).

I think this should be it:
```suggestion
    async def write_attributes(
        self,
        attributes: dict[str | int, Any],
        manufacturer: int | None = None,
        **kwargs,
    ) -> list:
```